Src/FluentAssertions/Common/MemberPath.cs

```diff
@@ -55,18 +57,26 @@ private bool IsParentOf(MemberPath candidate)
             string[] candidateSegments = candidate.Segments;
 
             return candidateSegments.Length > Segments.Length &&
-                   candidateSegments.Take(Segments.Length).SequenceEqual(Segments);
+                   candidateSegments.Take(Segments.Length).SequenceEqual(Segments, MemberPathSegmentEqualityComparer);
         }
 
         private bool IsChildOf(MemberPath candidate)
         {
             string[] candidateSegments = candidate.Segments;
 
             return candidateSegments.Length < Segments.Length
-                   && candidateSegments.SequenceEqual(Segments.Take(candidateSegments.Length));
+                   && candidateSegments.SequenceEqual(Segments.Take(candidateSegments.Length),
+                       MemberPathSegmentEqualityComparer);
+        }
+
+        public MemberPath Extend(MemberPath nextPath, string separator = ".")
```

Comment:
🔧 Isn't `Combine` a better name? That's what .NET uses for combining file paths.

Reply:
✅ Renamed to `Combine`